beacon-chain/blockchain/receive_attestation.go

```diff
@@ -43,15 +49,15 @@ func (s *Service) ReceiveAttestationNoPubsub(ctx context.Context, att *ethpb.Att
 }
 
 // AttestationPreState returns the pre state of attestation.
-func (s *Service) AttestationPreState(ctx context.Context, att *ethpb.Attestation) (iface.BeaconState, error) {
-	ss, err := helpers.StartSlot(att.Data.Target.Epoch)
+func (s *Service) AttestationTargetState(ctx context.Context, attTarget *ethpb.Checkpoint) (iface.BeaconState, error) {
```

Comment:
Changes like this might be better suited in another PR. They are general enough and can reduce the overall PR delta.